### PR TITLE
Sanitize user input for shell commands

### PR DESCRIPTION
**Pull request by emilyzheng**
Convert user input to shell-escaped strings so they can be safely used in a shell command line.

### CODE REVIEW

**Review by midnightercz — APPROVED** [review]

**Comment by hluk** [comment]
+1 Nice. :)

File: pubtools/_quay/command_executor.py [hunk 1]
```diff
@@ -46,7 +47,7 @@ def skopeo_login(self, username=None, password=None):
 
         cmd_login = (
             "skopeo login --authfile $HOME/.docker/config.json -u {0} --password-stdin quay.io"
-        ).format(username)
+        ).format(pipes.quote(username))
```
Comment (querti) [inline pubtools/_quay/command_executor.py:50]: I wonder, is this safe for Python 3? The tests passed, but the documentation makes it seem like it's deprecated. Could you pls use the 'six' module for it instead?

File: pubtools/_quay/command_executor.py [hunk 2]
```diff
@@ -78,7 +79,7 @@ def tag_images(self, source_ref, dest_refs, all_arch=False):
             LOG.info(
                 "Tagging source '{0}' to destination '{1}'".format(source_ref, dest_ref)
             )
-            self._run_cmd(cmd.format(source_ref, dest_ref))
+            self._run_cmd(cmd.format(pipes.quote(source_ref), pipes.quote(dest_ref)))
```
Comment (querti) [inline pubtools/_quay/command_executor.py:82]: Would it be possible to move the sanitization to RemoteExecutor's _run_cmd? That way we won't have to think about it when we add new commands.

**Review by querti — CHANGES_REQUESTED** [review]

File: pubtools/_quay/command_executor.py (hunk 1 above)
Comment (emilyzheng) [reply pubtools/_quay/command_executor.py]: Yeah, fixed.

File: pubtools/_quay/command_executor.py (hunk 2 above)
Comment (emilyzheng) [reply pubtools/_quay/command_executor.py]: The quote needs to be applied to user input. RemoteExecutor gets the whole command string so which part is from input can't be identified.

**Review by querti — APPROVED** [review]

File: pubtools/_quay/command_executor.py (hunk 1 above)
Comment (twaugh) [reply pubtools/_quay/command_executor.py]: There's probably a more straightforward way to do this. We're quoting a string and substituting it into another string, then shlex.splitting it before passing it to subprocess.Popen. Maybe it should just be a list of strings to start with?

Comment (querti) [reply pubtools/_quay/command_executor.py]: The reason for using quote is that remote execution is done with 'paramiko' library, which doesn't support splitting the input (AFAIK)

Comment (twaugh) [reply pubtools/_quay/command_executor.py]: Ah, ok.

**Pull request merged**